hooks/use-popover/index.js

```diff
@@ -1,15 +1,15 @@
 import { Popover } from '@wordpress/components';
-import { useState } from '@wordpress/element';
+import { useState, useCallback, useMemo } from '@wordpress/element';
 import { useOnClickOutside } from '../use-on-click-outside';
 
 export const usePopover = () => {
 	// Use internal state instead of a ref to make sure that the component
 	// re-renders when the popover's anchor updates.
 	const [popoverAnchor, setPopoverAnchor] = useState();
 	const [isVisible, setIsVisible] = useState(false);
-	const toggleVisible = () => {
-		setIsVisible(!isVisible);
-	};
+	const toggleVisible = useCallback(() => {
+		setIsVisible((visible) => !visible);
```

Comment:
This here is also cool. I always forget that you can pass an updater function to the state setter and get access to the current value...

Reply:
@kirtangajjar there's a difference!

The setter function from `useState` has two modes:

- One in which you pass the actual value you want to set.
- One in which you pass a function. In this scenario (the one I updated to)  the function gets the current value as an argument and must return the new value you want to set. By doing this we ensure no external dependencies so the callback can remain fully pure. 

I hope I clarified here!